### ISSUE TITLE
uninitialized constant Capistrano::SCM::Git::StringIO

### ISSUE THREAD

**Issue by jurgens** [capistrano/capistrano#2163]
```
> cap staging deploy
00:00 git:wrapper
      01 mkdir -p /tmp
    ✔ 01 deploy@_____ 0.432s
#<Thread:0x000000011dcb1ec0 /Users/jurgen/.rbenv/versions/3.3.4/lib/ruby/gems/3.3.0/gems/sshkit-1.23.2/lib/sshkit/runners/parallel.rb:10 run> terminated with exception (report_on_exception is true):
/Users/jurgen/.rbenv/versions/3.3.4/lib/ruby/gems/3.3.0/gems/sshkit-1.23.2/lib/sshkit/runners/parallel.rb:15:in `rescue in block (2 levels) in execute': Exception while executing as deploy@18.222.44.78: uninitialized constant Capistrano::SCM::Git::StringIO (SSHKit::Runner::ExecuteError)
	from /Users/jurgen/.rbenv/versions/3.3.4/lib/ruby/gems/3.3.0/gems/sshkit-1.23.2/lib/sshkit/runners/parallel.rb:11:in `block (2 levels) in execute'
/Users/jurgen/.rbenv/versions/3.3.4/lib/ruby/gems/3.3.0/gems/capistrano-3.19.1/lib/capistrano/scm/tasks/git.rake:9:in `block (3 levels) in eval_rakefile': uninitialized constant Capistrano::SCM::Git::StringIO (NameError)
	from /Users/jurgen/.rbenv/versions/3.3.4/lib/ruby/gems/3.3.0/gems/sshkit-1.23.2/lib/sshkit/backends/abstract.rb:31:in `instance_exec'
	from /Users/jurgen/.rbenv/versions/3.3.4/lib/ruby/gems/3.3.0/gems/sshkit-1.23.2/lib/sshkit/backends/abstract.rb:31:in `run'
	from /Users/jurgen/.rbenv/versions/3.3.4/lib/ruby/gems/3.3.0/gems/sshkit-1.23.2/lib/sshkit/runners/parallel.rb:12:in `block (2 levels) in execute'
(Backtrace restricted to imported tasks)
cap aborted!
SSHKit::Runner::ExecuteError: Exception while executing as deploy@18.222.44.78: uninitialized constant Capistrano::SCM::Git::StringIO (SSHKit::Runner::ExecuteError)


Caused by:
NameError: uninitialized constant Capistrano::SCM::Git::StringIO (NameError)
```
---

### Steps to reproduce
I've upgraded rails app to rails 7.1.5 as well as some dependencies
at the end I figured that it was psych 5.2.0 which was causing this error
downgrade to version 5.1.2 helped

### Expected behavior
Tell us what should happen

### Actual behavior
Tell us what happens instead

### System configuration
```
Environment

    Ruby     ruby 3.3.4 (2024-07-09 revision be1089c8ec) [arm64-darwin24]
    Rubygems 3.5.11
    Bundler  N/A
    Command  /Users/jurgen/.rbenv/versions/3.3.4/bin/cap staging doctor

Gems

    capistrano           3.19.1
    airbrussh            1.5.3
    rake                 13.2.1
    sshkit               1.23.2
    net-ssh              7.3.0
    capistrano-bundler   2.1.1
    capistrano-passenger 0.2.1
    capistrano-rails     1.6.3
    capistrano-rbenv     2.2.0
    capistrano-sidekiq   2.3.1

Variables

    :application                     "hubblefinancial"
    :assets_manifests                [#<Pathname:/home/deploy/hubblefinancial_staging/current/public/assets/.sprockets-manifest*>, #<Pathname:/home/deploy/hubblefinancial_staging/current/public/assets/manifest*.*>]
    :assets_prefix                   "assets"
    :assets_roles                    [:web]
    :branch                          "remove-stripe"
    :bundle_bins                     ["gem", "rake", "rails", "sidekiq", "sidekiqctl"]
    :bundle_binstubs                 nil
    :bundle_binstubs_command         :install
    :bundle_check_before_install     true
    :bundle_clean_options            ""
    :bundle_config                   {:deployment=>true}
    :bundle_env_variables            {}
    :bundle_flags                    "--quiet"
    :bundle_gemfile                  nil
    :bundle_jobs                     4
    :bundle_path                     #<Pathname:/home/deploy/hubblefinancial_staging/shared/bundle>
    :bundle_roles                    :all
    :bundle_servers                  [#<Capistrano::Configuration::Server:0x000000011ec3d268 @keys=[], @local=false, @user="deploy", @hostname="18.222.44.78", @port=nil, @properties=#<Capistrano::Configuration::Server::Properties:0x000000011ec7ed08 @properties={}, @roles=#<Set: {:app, :db, :web}>>>]
    :bundle_without                  "development:test"
    :chruby_map_bins                 ["sidekiq", "sidekiqctl"]
    :conditionally_migrate           false
    :default_env                     {:rbenv_root=>"$HOME/.rbenv", :rbenv_version=>"3.3.4"}
    :deploy_to                       "/home/deploy/hubblefinancial_staging"
    :deploy_via                      :remote_cache
    :format                          :airbrussh
    :git_environmental_variables     {:git_askpass=>"/bin/echo", :git_ssh=>"/tmp/git-ssh-894e67d094bf2a583cb6.sh"}
    :git_max_concurrent_connections  10
    :git_shallow_clone               false
    :git_wait_interval               0
    :git_wrapper_path                "/tmp/git-ssh-894e67d094bf2a583cb6.sh"
    :keep_assets                     nil
    :keep_releases                   5
    :linked_dirs                     ["log", "tmp/pids", "tmp/cache", "tmp/sockets", "tmp/data", "public/system", "public/assets"]
    :linked_files                    ["config/database.yml", "config/master.key", ".env"]
    :local_user                      "jurgen"
    :log_level                       :debug
    :migration_command               "db:migrate"
    :migration_role                  :db
    :migration_servers               #<Capistrano::Configuration::Server:0x000000011ec3d268 @keys=[], @local=false, @user="deploy", @hostname="18.222.44.78", @port=nil, @properties=#<Capistrano::Configuration::Server::Properties:0x000000011ec7ed08 @properties={}, @roles=#<Set: {:app, :db, :web}>>>
    :passenger_environment_variables {}
    :passenger_in_path               true
    :passenger_restart_command       "passenger-config restart-app"
    :passenger_restart_limit         2
    :passenger_restart_options       "/home/deploy/hubblefinancial_staging --ignore-app-not-running"
    :passenger_restart_runner        :sequence
    :passenger_restart_wait          5
    :passenger_restart_with_sudo     false
    :passenger_roles                 :app
    :passenger_rvm_ruby_version      nil
    :pty                             false
    :rails_env                       "staging"
    :rbenv_map_bins                  ["rake", "gem", "bundle", "ruby", "rails"]
    :rbenv_path                      "$HOME/.rbenv"
    :rbenv_prefix                    "$HOME/.rbenv/bin/rbenv exec"
    :rbenv_roles                     :all
    :rbenv_ruby                      "3.3.4"
    :rbenv_ruby_dir                  "$HOME/.rbenv/versions/3.3.4"
    :rbenv_type                      :user
    :repo_url                        "git@github.com:HubbleFinancial/hubble-financial-app.git"
    :rvm_map_bins                    ["sidekiq", "sidekiqctl"]
    :sidekiq_default_hooks           true
    :sidekiq_enable_lingering        true
    :sidekiq_env                     "staging"
    :sidekiq_error_log               "/home/deploy/hubblefinancial_staging/shared/log/sidekiq.error.log"
    :sidekiq_lingering_user          nil
    :sidekiq_log                     "/home/deploy/hubblefinancial_staging/shared/log/sidekiq.log"
    :sidekiq_roles                   :app
    :sidekiq_service_templates_path  "config/deploy/templates"
    :sidekiq_service_unit_name       "sidekiq"
    :sidekiq_service_unit_user       :user
    :ssh_options                     {:forward_agent=>true}
    :stage                           :staging
    :tmp_dir                         "/tmp"

    :deploy_via is not a recognized Capistrano setting (/Users/jurgen/.rbenv/versions/3.3.4/lib/ruby/3.3.0/delegate.rb:87)

Servers (1)

    deploy@18.222.44.78 [:app, :db, :web]

```

**Comment by jurgens** [comment]
could it be this change?
https://github.com/ruby/psych/commit/47aecb4346f26e957612121e115fc1244572b6a6

**Comment by pawelkomarnicki** [comment]
You need to edit the gem and add:

`require 'stringio'`

in capistrano-3.19.1/lib/capistrano/scm/tasks/git.rake

**Comment by jurgens** [comment]
thanks guys
really a quick fix!

**Comment by jules-w2** [comment]
i confirm, @taketo1113' s PR  fix this issue.
I hope it can be merged quickly

**Comment by mattbrictson** [comment]
Fixed in capistrano [3.19.2](https://github.com/capistrano/capistrano/releases/tag/v3.19.2). ✅
